File: Contests/Codechef/AUG20A/CHEFWED.py
import sys 
# For getting input from input.txt file 
sys.stdin = open('input.txt', 'r')  
  
# Printing the Output to output.txt file 
sys.stdout = open('output.txt', 'w') 

for _ in range(int(input())):
    n, k = map(int, input().split())
    f = list(map(int, input().split()))
    i = 0
    if k == 1:
        l = []
        l1 = []
        while i < n:
            if f[i] not in l1:
                l1.append(f[i])
                if i == n - 1:
                    l.append(l1)
                    l1.clear()
                i += 1
            else:
                l.append(l1)
                l1.clear()
        k = len(l) * k
        l.clear()
        print(k, end="\n")
    else:
        coun = 0
        s = set(f)
        li = list(s)
        for i in range(len(li)):
            if f.count(li[i]) > 1:
                t = f.count(li[i])
                coun = coun + t
        k = k + coun
        li.clear()
        print(k, end="\n")

File: Contests/Codechef/AUG20A/CHEFWED.cpp
#include <bits/stdc++.h>
using namespace std;

#define ll long long
#define pb push_back
#define mp make_pair
#define ar array
#define ff first
#define ss second

int main() {
	ios_base::sync_with_stdio(0); cin.tie(0);

#ifndef ONLINE_JUDGE
    // for getting input from input.txt
    freopen("input.txt", "r+", stdin);
    // for writing output to output.txt
    freopen("output.txt", "w+", stdout);
#endif

    int t;
    cin >> t;
    while(t--) {
    	int n,k;
    	cin >> n >> k;
        // assert(k > 1);
    	int f[n+1];
    	map<int,int> cnt;
    	for(int i=0;i<n;i++) {
    		cin >> f[i];
    	}
    	vector <map<int,int>> tables;
        cnt[f[0]]++;
        tables.pb(cnt);
        for(int i=1;i<n;i++) {
            int currentAnswer = k;
            for(auto u: cnt) {
                if(u.ss > 1) {
                    currentAnswer += u.ss;
                } 
            }
            int val;
            if(cnt[f[i]] == 0) {
                val = 0;
            }
            else if(cnt[f[i]] == 1) {
                val = 2;
            }
            else {
                val = 1;
            }
            if(currentAnswer + val <= currentAnswer + k) {
                cnt[f[i]]++;
                tables.pop_back();
                tables.pb(cnt);
            }
            else {
                cnt.clear();
                cnt[f[i]]++;
                tables.pb(cnt);
            }
        }
        // for(int i=0;i<tables.size();i++) {
        //     cout << "table " << i + 1 << '\n';
        //     for(auto v : tables[i]) {
        //         cout << v.ff << " " << v.ss << '\n';
        //     }
        // }   
        ll ans1 = 0;
        for(auto u: tables) {
            ans1+=k;
            for(auto v : u) {
                if(v.ss > 1) {
                    ans1 += v.ss;
                }
            }
        }
    	cout << ans1 << '\n';
    }
}